File: java/Y.java
/*
* @Autor: Darwin Neira Carrasco
* @Email: dneirac@unsa.edu.pe
* @Descripcion:
*/

public class Y extends X {
  int y = 1;

  @Override
  public A foo(B a) {
    System.out.format("Y::foo(B) retorna B %d\n", y);
    return a;
  }
}

File: python/Y.py
'''
# @Author      : Darwin Neira Carrasco 
# @Email       : dneirac@unsa.edu.pe
# @File        : Y
#
# @Description : 
'''

from X import *

class Y(X):
    def foo(self,a):#A
        print(type(a))
        b = B()
        print("parametro: A, retorno: B")
        return b #B
